src/utils/index.ts

```diff
@@ -142,23 +142,36 @@ export function formatString(str: string, params?: object): string {
  * Generates the update mask for the provided object.
  * Note this will ignore the last key with value undefined.
  *
- * @param {[key: string]: any} obj The object to generate the update mask for.
- * @return {Array<string>} The computed update mask list.
+ * @param obj The object to generate the update mask for.
+ * @param terminalPaths The optional map of keys for maximum paths to traverse.
+ *      Nested objects beyond that path will be ignored. This is useful for
+ *      keys with variable object values.
+ * @param root The path so far.
+ * @return The computed update mask list.
  */
-export function generateUpdateMask(obj: {[key: string]: any}): string[] {
+export function generateUpdateMask(
+  obj: any, terminalPaths: Set<string> = new Set(), root = ''
```

Comment:
Set is a bit okverkill here.  Let's use a regular array and `terminalPaths.indexOf()` to check membership. This array will be very short in practice, so performance shouldn't be an issue.

Reply:
I think `Set` would have been ok. It is pretty lightweight and readily available to us as we are using ES6. But since the array won't grow too large, I switched to `string[]`.